### ISSUE TITLE
Source Locators are not striped if using firrtlCircuit and infoMode ignore

### ISSUE THREAD

**Issue by jackkoenig** [freechipsproject/firrtl#1013]
This comes up when using chisel3.Driver.execute

**Type of issue**: bug report

**What is the use case for changing the behavior?**

Ignoring info should mean stripping source locators. It doesn't seem to work if you pass a circuit instead of something to parse.

We should also check that it works with Protobuf parsing

**Impact**: no functional change

**Development Phase**: request

**Other information** (e.g. detailed explanation, stacktraces, related issues, suggestions how to fix, links for us to have context, eg. Stack Overflow, gitter, etc)

**Comment by seldridge** [comment]
Yeah, an actual circuit will sneak past. As will a `FirrtlCircuitAnnotation`. In the context of #1005, that would mean a `Phase` that could run as early as you want to strip info from `FirrtlCircuitAnnotation`s if the current info mode is "ignore". That would then require that `proto.FromProto.fromFile` be modified to take the info mode as a parameter.

Alternatively, the same `Phase` could run after `AddCircuit` and just strip info mode if needed.

Relatedly, it's somewhat annoying how the info mode and the circuit have a dependency on each other. You can't determine one without the other...
